File: InMotionGIT.Common/Extensions/DbDataReaderExtensions.vb
Imports System.Data.Common
Imports System.Runtime.CompilerServices

Namespace Extensions

    ''' <summary>
    ''' Extension methods for the DbDataReader type
    ''' </summary>
    Public Module DbDataReaderExtensions

        <Extension()>
        Public Function StringValue(ByVal value As DbDataReader, ordinal As Integer) As String
            Dim result As String = String.Empty
            If Not value.IsDBNull(ordinal) Then
                result = value.GetString(ordinal).Trim
            End If
            Return result
        End Function

        <Extension()>
        Public Function DateTimeValue(ByVal value As DbDataReader, ordinal As Integer) As Date
            Dim result As Date = Date.MinValue
            If Not value.IsDBNull(ordinal) Then
                result = value.GetDateTime(ordinal)
            End If
            Return result
        End Function

        <Extension()>
        Public Function NumericValue(ByVal value As DbDataReader, ordinal As Integer) As Decimal
            Dim result As Decimal = 0
            If Not value.IsDBNull(ordinal) Then
                result = value.GetDecimal(ordinal)
            End If
            Return result
        End Function

        <Extension()>
        Public Function BooleanValue(ByVal value As DbDataReader, ordinal As Integer) As Boolean
            Dim result As Boolean = 0
            If Not value.IsDBNull(ordinal) Then
                result = (value.GetDecimal(ordinal) = 1)
            End If
            Return result
        End Function

    End Module

End Namespace

File: Common/InMotionGIT.Common/Extensions/DbDataReaderExtensions.cs
using System;
using System.Data.Common;
using Microsoft.VisualBasic.CompilerServices;

namespace InMotionGIT.Common.Extensions
{

    /// <summary>
    /// Extension methods for the DbDataReader type
    /// </summary>
    public static class DbDataReaderExtensions
    {

        public static string StringValue(this DbDataReader value, int ordinal)
        {
            string result = string.Empty;
            if (!value.IsDBNull(ordinal))
            {
                result = value.GetString(ordinal).Trim();
            }
            return result;
        }

        public static DateTime DateTimeValue(this DbDataReader value, int ordinal)
        {
            var result = DateTime.MinValue;
            if (!value.IsDBNull(ordinal))
            {
                result = value.GetDateTime(ordinal);
            }
            return result;
        }

        public static decimal NumericValue(this DbDataReader value, int ordinal)
        {
            decimal result = 0m;
            if (!value.IsDBNull(ordinal))
            {
                result = value.GetDecimal(ordinal);
            }
            return result;
        }

        public static bool BooleanValue(this DbDataReader value, int ordinal)
        {
            bool result = Conversions.ToBoolean(0);
            if (!value.IsDBNull(ordinal))
            {
                result = value.GetDecimal(ordinal) == 1m;
            }
            return result;
        }

    }

}